int32 unique_wp_num
Coords[] data


